pkg/ipmi/ipmi.go

```diff
@@ -216,3 +222,46 @@ func (i *IPMI) SetSystemFWVersion(version string) error {
 
 	return nil
 }
+
+func (i *IPMI) getDeviceID() ([]byte, error) {
```

Comment:
Should this return a struct similar to this instead:
https://github.com/ipmitool/ipmitool/blob/master/include/ipmitool/ipmi_mc.h#L68

Reply:
And in that way avoid the `data[6]&_ADTL_SEL_DEVICE` outside of this function.

Reply:
Apart from this it lgtm.

Reply:
Updated for struct return.